types/hapipal__toys/index.d.ts

```diff
@@ -0,0 +1,102 @@
+export function onPreResponse(method: Lifecycle.Method, options?: ServerExtOptions): ServerExtEventsObject;
+export function onPostResponse(method: Lifecycle.Method, options?: ServerExtOptions): ServerExtEventsObject;
+
+export type AssignRouteDefaults = (routes: Partial<ServerRoute> | Array<Partial<ServerRoute>>) => ServerRoute | ServerRoute[];
+export function withRouteDefaults(defaults: Partial<ServerRoute>): AssignRouteDefaults;
+
+export interface ToysPreShorthand {
+    [assignKey: string]: Lifecycle.Method;
+}
+export type ToysPreArg = (ToysPreShorthand | Lifecycle.Method);
+export function pre(options: ToysPreArg | ToysPreArg[]): RouteOptionsPreArray;
+
+export interface ReacherOptions {
+    separator?: string;
+    default?: any;
+    strict?: boolean;
+    functions?: boolean;
+    iterables?: boolean;
+}
+export type PerformReach = (input?: object) => any;
+export function reacher(chain: string | Array<(string | number)>, options?: ReacherOptions): PerformReach;
+
+export type PerformTransform = (input: object) => object;
+export interface Transformer {
+    [fromPath: string]: string;
+}
+export function transformer(transform: Transformer, options?: ReacherOptions): PerformTransform;
+
+export const noop: Plugin<any>;
+
+export function header(response: ResponseObject | Boom, name: string, value: string, options?: ResponseObjectHeaderOptions): void;
+export function getHeaders(response: ResponseObject | Boom): { [header: string]: string };
+
+export function code(response: ResponseObject | Boom, code: number): void;
+export function getCode(response: ResponseObject | Boom): number;
+
+export namespace auth {
+    type ServerAuthSchemeAuthenticate = (request: Request, h: ResponseToolkit) => Lifecycle.ReturnValue;
+    function strategy(server: Server, name: string, authenticate: ServerAuthSchemeAuthenticate): ServerAuthScheme;
+}
+
+export interface EventOptions {
+    multiple?: boolean;
+    error?: boolean;
+}
+// has to resolve an `any` because return type is dependent on the event that is emitted
+export function event(emitter: EventEmitter, eventName: string, options?: EventOptions): Promise<any>;
+
+export interface StreamOptions {
+    cleanup?: boolean;
+}
+export function stream(stream: Stream, options?: StreamOptions & FinishedOptions): Promise<void>;
+
+export type TypesWithRealmsAndOptions = (Server | Request | ResponseToolkit | ServerRealm | ServerRoute);
+export function options(obj: TypesWithRealmsAndOptions): object;
+export function realm(obj: TypesWithRealmsAndOptions): object;
+
+export function rootRealm(realm: ServerRealm): ServerRealm;
+
+export function state(realm: ServerRealm, pluginName: string): object;
+export function rootState(realm: ServerRealm, pluginName: string): object;
+
+export type AncestorRealmIterator = (realm: ServerRealm) => void;
+export function forEachAncestorRealm(realm: ServerRealm, fn: AncestorRealmIterator): void;
+
+export function asyncStorage(identifier: string): any;
```

Comment:
Perhaps it could be interesting to use a generic here too instead of `any` as the return type?

Reply:
I had originally written this as `export function asyncStorage<T>(identifier: string): T;` which worke better, but the linter complained about a single use generic.

Reply:
https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md at the bottom is the case mentioned. I could disable that rule if that seems better, as I agree that the `any` is a little weird in this use case.